src/providers/appium.ts

```diff
@@ -200,3 +201,74 @@ export function getAppBundleId(path: string): Promise<string> {
     });
   });
 }
+
+async function getLatestBuildToolsVersion(
+  androidHome: string,
+): Promise<string | undefined> {
+  return new Promise((resolve, reject) => {
+    const buildToolsPath = path.join(androidHome, "build-tools");
+
+    fs.readdir(buildToolsPath, (err, files) => {
+      if (err) {
+        console.error(`getLatestBuildToolsVersion: ${err}`);
+        return reject("Error reading build-tools directory");
+      }
+
+      // Filter out files that are not valid build-tools versions (directories with version numbers)
+      const versions = files.filter((file) =>
+        /^\d+\.\d+\.\d+(-rc\d+)?$/.test(file),
+      );
+
+      if (versions.length === 0) {
+        return reject("No valid build-tools version found");
+      }
+
+      const latestVersion = versions.sort((a, b) => (a > b ? -1 : 1))[0];
```

Comment:
this is a sorting function 

(a, b) => (a > b ? -1 : 1)

better to keep this in utils and name the sorter